tests/checkpoint/test_checkpoint.py

```diff
@@ -3718,3 +3719,686 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
+    )
+    runtime_batch_request = RuntimeBatchRequest(
+        **{
+            "datasource_name": "my_datasource",
+            "data_connector_name": "my_runtime_data_connector",
+            "data_asset_name": "test_df",
+            "batch_identifiers": {
+                "pipeline_stage_name": "core_processing",
+                "airflow_run_id": 1234567890,
+            },
+            "runtime_parameters": {"batch_data": test_df},
+        }
+    )
+
+    # add checkpoint config
+    checkpoint_config = {
+        "class_name": "Checkpoint",
+        "name": "my_checkpoint",
+        "config_version": 1,
+        "run_name_template": "%Y-%M-foo-bar-template",
+        "expectation_suite_name": "my_expectation_suite",
+        "action_list": [
+            {
+                "name": "store_validation_result",
+                "action": {
+                    "class_name": "StoreValidationResultAction",
+                },
+            },
+            {
+                "name": "store_evaluation_params",
+                "action": {
+                    "class_name": "StoreEvaluationParametersAction",
+                },
+            },
+            {
+                "name": "update_data_docs",
+                "action": {
+                    "class_name": "UpdateDataDocsAction",
+                },
+            },
+        ],
+        "validations": [{"batch_request": batch_request}],
+    }
+
+    context.add_checkpoint(**checkpoint_config)
+
+    results = context.run_checkpoint(checkpoint_name="my_checkpoint")
+    assert results["success"] == False
+    assert len(results.run_results.values()) == 1
+    assert (
+        list(results.run_results.values())[0]["validation_result"]["statistics"][
+            "evaluated_expectations"
+        ]
+        == 1
+    )
+    assert (
+        list(results.run_results.values())[0]["validation_result"]["statistics"][
+            "successful_expectations"
+        ]
+        == 0
+    )
+
+    results = context.run_checkpoint(
+        checkpoint_name="my_checkpoint",
+        validations=[{"batch_request": runtime_batch_request}],
+    )
+    assert results["success"] == False
+    assert len(results.run_results.values()) == 2
+    assert (
+        list(results.run_results.values())[0]["validation_result"]["statistics"][
+            "evaluated_expectations"
+        ]
+        == 1
+    )
+    assert (
+        list(results.run_results.values())[0]["validation_result"]["statistics"][
+            "successful_expectations"
+        ]
+        == 0
+    )
+    assert (
+        list(results.run_results.values())[1]["validation_result"]["statistics"][
+            "evaluated_expectations"
+        ]
+        == 1
+    )
+    assert (
+        list(results.run_results.values())[1]["validation_result"]["statistics"][
+            "successful_expectations"
+        ]
+        == 1
+    )
+
+
+def test_newstyle_checkpoint_does_not_pass_dataframes_via_batch_request_into_checkpoint_store(
```

Comment:
An error should be raised if an attempt is made to save DataFrames to `CheckpointStore`.